test/messagebus/progress_outbox_with_proof.js

```diff
@@ -0,0 +1,274 @@
+				await messageBusUtils.progressOutboxWithProof(params, false);
+
+			});
+
+			it('should fail when message status is undeclared and messagehash status in outbox is declared', async () => {
+
+				await messageBusUtils.declareMessage(params, true);
+				params.messageStatus = MessageStatusEnum.Undeclared;
+				await messageBusUtils.progressOutboxWithProof(params, false);
+
+			});
+
+			it('should fail when message status is undeclared and messagehash status in outbox is progressed', async () => {
+
+				await messageBusUtils.declareMessage(params, true);
+				params.messageStatus = MessageStatusEnum.Declared;
+				await messageBusUtils.progressOutboxWithProof(params, true);
+				
+				params.messageStatus = MessageStatusEnum.Undeclared;
+				await messageBusUtils.progressOutboxWithProof(params, false);
+				
+			});
+
+			it('should fail when message status is undeclared and messagehash status in outbox is declared revocation', async () => {
+				
+				await messageBusUtils.declareMessage(params, true);
+				await messageBusUtils.declareRevocationMessage(params, true);
+				
+				params.messageStatus = MessageStatusEnum.Undeclared;
+				await messageBusUtils.progressOutboxWithProof(params, false);
+
+			});
+
+			it('should fail when message status is undeclared and messagehash status in outbox is revoked', async () => {
+				
+				await messageBusUtils.declareMessage(params, true);
+				await messageBusUtils.declareRevocationMessage(params, true);
+				params.messageStatus = MessageStatusEnum.DeclaredRevocation;
+				await messageBusUtils.progressOutboxRevocation(params, true);
+				
+				params.messageStatus = MessageStatusEnum.Undeclared;
+				await messageBusUtils.progressOutboxWithProof(params, false);
+
+			});
+			
+			it('should fail when message status is declared revocation and messagehash status in outbox is undeclared', async () => {
+
+				params.messageStatus = MessageStatusEnum.DeclaredRevocation;
+				await messageBusUtils.progressOutboxWithProof(params, false);
+
+			});
+
+			it('should fail when message status is declared revocation and messagehash status in outbox is declared', async () => {
+				
+				await messageBusUtils.declareMessage(params, true);
+				
+				params.messageStatus = MessageStatusEnum.DeclaredRevocation;
+				await messageBusUtils.progressOutboxWithProof(params, false);
+
+			});
+
+			it('should fail when message status is declared revocation and messagehash status in outbox is progressed', async () => {
+				
+				await messageBusUtils.declareMessage(params, true);
+				params.messageStatus = MessageStatusEnum.Declared;
+				await messageBusUtils.progressOutboxWithProof(params, true);
+				
+				params.messageStatus = MessageStatusEnum.DeclaredRevocation;
+				await messageBusUtils.progressOutboxWithProof(params, false);
+
+			});
+
+			it('should fail when message status is declared revocation and messagehash status in outbox is declared revocation', async () => {
```

Comment:
`It` statements for function with proof are very confusing. 
Please correct the`it` statement and use `source` and `target` terminology.